cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server.go

```diff
@@ -104,36 +114,38 @@ func (s *Server) GetAvailablePackages(ctx context.Context, request *corev1.GetAv
 
 	responsePackages := []*corev1.AvailablePackage{}
 	for _, pkgUnstructured := range pkgs.Items {
-		pkg := &corev1.AvailablePackage{}
-		name, found, err := unstructured.NestedString(pkgUnstructured.Object, "spec", "publicName")
-		if err != nil || !found {
-			return nil, status.Errorf(codes.Internal, "required field publicName not found on kapp-controller package: %v:\n%v", err, pkgUnstructured.Object)
+		pkg, err := availablePackageFromUnstructured(&pkgUnstructured)
+		if err != nil {
+			return nil, err
 		}
-		pkg.Name = name
-
-		version, found, err := unstructured.NestedString(pkgUnstructured.Object, "spec", "version")
-		if err != nil || !found {
-			return nil, status.Errorf(codes.Internal, "required field version not found on kapp-controller package: %v:\n%v", err, pkgUnstructured.Object)
-		}
-		pkg.Version = version
 		responsePackages = append(responsePackages, pkg)
 	}
 	return &corev1.GetAvailablePackagesResponse{
 		Packages: responsePackages,
 	}, nil
 }
 
-// GetPackageRepositories returns the package repositories based on the request.
-func (s *Server) GetPackageRepositories(ctx context.Context, request *corev1.GetPackageRepositoriesRequest) (*corev1.GetPackageRepositoriesResponse, error) {
-	// TODO: replace incluster config with the user config using token from request meta.
-	config, err := rest.InClusterConfig()
-	if err != nil {
-		return nil, fmt.Errorf("unable to create incluster config: %w", err)
+func availablePackageFromUnstructured(ap *unstructured.Unstructured) (*corev1.AvailablePackage, error) {
```

Comment:
Extracted this so that the functionality can be unit-tested in isolation, rather than as part of the `GetAvailablePackages` unit tests.